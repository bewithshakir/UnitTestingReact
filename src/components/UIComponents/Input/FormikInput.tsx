import { useField } from 'formik';

import Input, { InputProps } from "./Input";

export default function FormikInput(props: InputProps) {
    const [, meta] = useField(props as any);
    return <Input {...props}
        id={props.name}
        error={Boolean(meta.error && meta.touched)}
        helperText={meta.touched && meta.error ? meta.error : undefined}
    />;
}
