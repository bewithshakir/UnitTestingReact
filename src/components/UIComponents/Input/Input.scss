.MuiInputBase-input.Mui-disabled {
    background: #F7F7F7 0% 0% no-repeat padding-box;
    opacity: 1;
}

.MuiFormLabel-root{
    color: #404040 !important;
}

.MuiFormLabel-root.Mui-focused {
    color: #404040!important;
}

.MuiInputBase-input:invalid {
    box-shadow: none;
    border: 2px solid #DD1D21;
}

.MuiFormHelperText-error{
    color:#DD1D21 !important;
}

.super{
    font-size: 1em;
}