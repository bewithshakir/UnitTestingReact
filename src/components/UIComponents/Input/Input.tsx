import { Fragment } from 'react';
import { InputLabel, InputBase, FormControl, FormHelperText } from '@material-ui/core';
import './Input.scss';

interface props {
  name?: string;
  id?: string;
  width?: number | string;
  label: string;
  placeholder?: string;
  description?: string;
  disabled?: boolean;
  required?: boolean;
  autoComplete?: string;
  classes?: any;
  multiline?: boolean;
  type?: string;
  error?: boolean;
  value?: string;
  helperText?: string
  onChange?: (...args: any[]) => void;
  onBlur?: (...args: any[]) => void;
  onFocus?:(...args: any[]) => void;
  onClick?:(...args: any[]) => void;
  innerRef?:any;
  onKeyDown?:(...args: any[]) => void;
  inputProps?:any;
  endAdornment?:any;
  startAdornment?:any;
}

export default function Input (props: props) {

  return (
    <Fragment>
      <FormControl className='formInput'>
        <InputLabel shrink htmlFor={props.id} style={{ color: 'var(--Darkgray)' }} aria-labelledby={props.label} aria-required={props.required}>
          <b>{props.label.toUpperCase()}{props.required && props.label && (<span className='super'>*</span>)}</b>
        </InputLabel>
        <InputBase
          id={props.id}
          ref={props.innerRef} 
          style={{ width: props.width }}
          aria-describedby={props.description}
          name={props.name}
          value={props.value}
          disabled={props.disabled}
          placeholder={props.placeholder}
          aria-required={props.required}
          autoComplete={props.autoComplete}
          classes={props.classes}
          multiline={props.multiline}
          type={props.type}
          error={props.error}
          onChange={props.onChange}
          onBlur={props.onBlur}
          onClick={props.onClick}
          onKeyDown={props.onKeyDown}
          inputProps={props.inputProps}
          endAdornment={props.endAdornment}
          startAdornment={props.startAdornment}
        />
        {props.helperText && (
          <FormHelperText
            id={props.description}
            error={props.error}>
            {props.helperText}
          </FormHelperText>)}
      </FormControl>
    </Fragment>
  );
}

Input.defaultProps = {
  width: '100%',
  disabled: false,
  required: false,
  id: "input",
  autoComplete: "new-password",
  error: false,
};
