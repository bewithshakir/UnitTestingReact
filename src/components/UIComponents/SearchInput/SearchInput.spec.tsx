import React from 'react';
import { shallow } from 'enzyme';
import Input from './SearchInput';

describe('Rendering of Search Input Component', () => {

    const component = shallow(<Input/>);

    it('Search Input Snapshot testing', () => {
        expect(component).toBeDefined();
        expect(component).toMatchSnapshot();
    });

});