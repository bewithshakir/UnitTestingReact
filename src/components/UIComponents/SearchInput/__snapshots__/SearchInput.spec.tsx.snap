// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering of Search Input Component Search Input Snapshot testing 1`] = `
<ForwardRef(OutlinedInput)
  className="searchinput"
  id="SearchInput"
  inputProps={
    Object {
      "aria-label": "search",
    }
  }
  onChange={[Function]}
  placeholder="Search"
  startAdornment={
    <ForwardRef(InputAdornment)
      className="adornment"
      position="start"
    >
      <ForwardRef(IconButton)
        aria-label="search"
        edge="start"
      >
        <DropSearchIcon />
      </ForwardRef(IconButton)>
    </ForwardRef(InputAdornment)>
  }
  style={
    Object {
      "width": "100%",
    }
  }
  type="text"
  value=""
/>
`;
